### PR TITLE
Cleanup and Refactor MapActivity

### PR DESCRIPTION
**Pull request by Sporarum**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/MainApplication.kt [hunk 1]
```diff
@@ -16,13 +16,4 @@ class MainApplication : Application() {
             return instance!!.applicationContext
         }
     }
-
-    override fun onCreate() {
-        super.onCreate()
-        // initialize for any
-
-        // Use ApplicationContext.
```
Comment (Ph0tonic) [inline app/src/main/java/ch/epfl/sdp/MainApplication.kt:24]: We should move `Mapbox.getInstance(...)` in here.

File: app/src/main/java/ch/epfl/sdp/MapActivity.kt [hunk 2]
```diff
@@ -195,33 +184,13 @@ class MapActivity : MapViewBaseActivity(), OnMapReadyCallback {
...
             }
 
-            heatmapGeoJsonSource = GeoJsonSource(getString(R.string.heatmap_source_ID), GeoJsonOptions()
-                    .withCluster(true)
-                    .withClusterProperty("intensities", Expression.literal("+"), Expression.get("intensity"))
-                    .withClusterMaxZoom(13)
-            )
-            heatmapGeoJsonSource.setGeoJson(FeatureCollection.fromFeatures(heatmapFeatures))
-            style.addSource(heatmapGeoJsonSource)
+            setupHeatMap(style)
```
Comment (Ph0tonic) [inline app/src/main/java/ch/epfl/sdp/MapActivity.kt:187]: This change is useless because we extract it on our branch and will juste create some conflicts.

Comment (Sporarum) [reply app/src/main/java/ch/epfl/sdp/MapActivity.kt:187]: Okay, will fix that

File: app/src/main/java/ch/epfl/sdp/MainApplication.kt (hunk 1 above)
Comment (Sporarum) [reply app/src/main/java/ch/epfl/sdp/MainApplication.kt:24]: Done

**Pull request merged**